src/main.rs

```diff
@@ -17,14 +17,26 @@ impl ZellijPlugin for State {
     fn pipe(&mut self, pipe_message: PipeMessage) -> bool {
         let session_name = pipe_message.payload.unwrap().to_string();
         let collection: Vec<&str> = session_name.split("::").collect::<Vec<&str>>().clone();
+
         let session_name = collection[0];
-        let layout_name = "default";
-        let layout: LayoutInfo = LayoutInfo::File(layout_name.to_string());
         let mut cwd = None;
-        if collection.len() == 2 {
-            cwd = Some(PathBuf::from(collection[1]));
-        }
-        switch_session_with_layout(Some(&session_name), layout, cwd);
+
+        let layout_name = match collection.len() {
+            2 => {
+                // Original format: session::cwd
+                cwd = Some(PathBuf::from(collection[1]));
+                "default".to_string()
+            }
+            3 => {
+                // New format: session::cwd::layout
+                cwd = Some(PathBuf::from(collection[1]));
+                format!("{}.kdl", collection[2])
+            }
+            _ => "default".to_string(),
+        };
```

Comment:
I think you could handle all of this like before with 2 `if`s:

```
if collection.len() >= 2 {
        cwd = Some(PathBuf::from(collection[1]));
}
if collection.len() == 3 {
        layout_name = format!("{}.kdl", collection[2])
}
```

Reply:
Yeah, that's a bit simpler and does seem to work as intended.